### ISSUE TITLE
support kontalk network

### ISSUE THREAD

**Issue by pravi** [ChatSecure/ChatSecure-iOS#370]
See this issue for details https://github.com/loqui/im/issues/230

Currently kontalk is available only on android and if chatsecure adds support for it, it would be a big boost to easiest reliable encryption option (otr can't work with multiple clients or in asynchronous connection).

**Comment by chrisballinger** [comment]
Is this just Xmpp?

**Comment by pravi** [comment]
Yes, with small customization for registration. Since it uses phone number hash based username, it can auto discover contacts.

You'll need,
SASL EXTERNAL authentication with SSL certificate on either STARTTLS or direct SSL (the latter still has problems with Tigase though)
phone number based registration (a small extension to jabber:iq:register which I will document soon on the project's wiki) or importing an existing key from an Android device (which requires the jabber:iq:register process anyway)
a builtin "server list" for a given network (kontalk.net for now is the only one) for bootstrapping
OpenPGP encryption for messages
minimal key management for public keys

**Comment by webratte** [comment]
Hi, 
I'm don't know much about tech and coding!!

But I follow this discussion on LibreSignal issues:
https://github.com/LibreSignal/LibreSignal/issues/37#

Since there is going a discussion about federation XMPP messengers to beat the big ones, maybe you should think about to move this issue to a earlier milestone ;-)

**Comment by masvil** [comment]
Some leaders are discussing here too: https://github.com/anurodhp/Monal/issues/9 (maybe not the better place).

**Comment by daniele-athome** [comment]
Hello there! I'm Kontalk project leader. I would be very much interested in seeing this implemented, waiting for a proper iOS client. It would help users getting on board with Apple devices, which would increase popularity, which would bring more developers, and one of them could be an iOS one :-)

I don't have any experience in iOS development (nor the hardware), is there something I can do in the meantime? I mean something that needs to be done server-side or with documentation so that this feature can be implemented smoothly.

**Comment by Neustradamus** [comment]
Any news on it?
